app/components/NewsBanner.vue

```diff
@@ -1,124 +1,83 @@
-<template>
-<div>
-  <div class="banner" :class="[bannerExists ? 'show' : '']" @click="followLink()">
-    <img class="bg-image" :src="currentBanner.thumbnail" />
-    <div class="image-container">
-      <img class="shadow-image__left" :src="currentBanner.thumbnail" />
-      <img class="shadow-image__right" :src="currentBanner.thumbnail" />
-      <img class="main-image" :src="currentBanner.thumbnail" />
-    </div>
-    <div class="title-container">
-      <h3 class="title">{{ headerText }}</h3>
-      <p class="subheading">{{ currentBanner.subHeader }}</p>
-    </div>
-    <div class="cta-container">
-      <button class="button learn-more" :disabled="!bannerExists">{{ currentBanner.linkTitle }}</button>
-      <button class="dismiss-button" @click.stop="closeBanner()" :disabled="!bannerExists || processingClose">{{ $t('Dismiss') }}</button>
-    </div>
-  </div>
-</div>
-</template>
-
-<script lang="ts" src="./NewsBanner.vue.ts"></script>
-
-<style lang="less" scoped>
-@import '../styles/index';
-
-.banner {
-  width: 100%;
-  min-height: 0;
-  max-height: 0;
-  background-color: @teal;
-  color: @white;
-  display: flex;
-  position: relative;
-  align-content: center;
-  overflow: hidden;
-  transition: all 0.5s cubic-bezier(0, 1, 0, 1);
-}
-
-.banner.show {
-  min-height: 92px;
-  max-height: 140px;
-  .padding(2);
-  .transition();
-}
-
-.bg-image {
-  position: absolute;
-  opacity: 0.1;
-  height: 400%;
-  top: -100%;
-  left: -100px;
-  filter: blur(6px);
-}
-
-.image-container {
-  position: relative;
-  margin-left: 50px;
-  width: 80px;
-
-  img {
-    top: 50%;
-    transform: translate(0, -50%);
+<script lang="tsx">
```

Comment:
basically the reason for this is to allow it to be imported and used by our other vue components, if we had started out using only TSX it would be better to do it that way but this is the most painless way i could find to integrate it into our existing component structure. if we want to have TSX components which are only being used by other TSX components we can totally do it that way

Reply:
Hmm.  Let's chat.  That doesn't make a whole lot of sense to me.  AFAIK the compiled component should be equivalent whether it is written in a `.vue` file or a `.tsx` file.  `.vue` is just syntactic sugar.